interfaces/matlab_experimental/1D/Domain1D.m

```diff
@@ -0,0 +1,696 @@
+        %     Instance of class :mat:class:`Domain1D`.
+        % :return:
+        %     Integer flag denoting the location of the domain,
+        %     beginning with 1 at the left.
+        domainIndex
+
+        % Type of the domain. ::
+        %
+        %     >> i = d.domainType
+        %
+        % :param d:
+        %     Instance of class :mat:class:`Domain1D`.
+        % :return:
+        %     Integer flag denoting the domain type.
+        domainType
+
+        % Determines whether a domain is a flow. ::
+        %
+        %     >> a = d.isFlow
+        %
+        % :param d:
+        %     Instance of class :mat:class:`Domain1D`.
+        % :return:
+        %     True if the domain is a flow domain, and false otherwise.
+        isFlow
+
+        % Determines whether a domain is an inlet. ::
+        %
+        %     >> a = d.isInlet
+        %
+        % :param d:
+        %     Instance of class :mat:class:`Domain1D`.
+        % :return:
+        %     True if the domain is an inlet, and false otherwise.
+        isInlet
+
+        % Determines if a domain is a surface. ::
+        %
+        %     >> a = d.isSurface
+        %
+        % :param d:
+        %     Instance of class :mat:class:`Domain1D`.
+        % :return:
+        %     True if the domain is a surface, and false otherwise.
+        isSurface
+
+        % Mass flux. ::
+        %
+        %     >> mdot = d.massFlux
+        %
+        % :param d:
+        %     Instance of class :mat:class:`Domain1D`.
+        % :return:
+        %     The mass flux in the domain.
+        massFlux
+
+        % Number of components. ::
+        %
+        %     >> n = d.nComponents
+        %
+        % :param d:
+        %     Instance of class :mat:class:`Domain1D`.
+        % :return:
+        %     Number of variables at each grid point.
+        nComponents
+
+        % Get the number of grid points. ::
+        %
+        %     >> n = d.nPoints
+        %
+        % :param d:
+        %     Instance of class :mat:class:`Domain1D`.
+        % :return:
+        %     Integer number of grid points.
+        nPoints
+
+    end
+
+    methods
+        %% Domain1D Class Constructor.
+
+        function d = Domain1D(varargin)
+            % Create a :mat:class:`Domain1D` object. 
+
+            ctIsLoaded;
+
+            if nargin == 1
+                a = varargin{1};
+
+                if strcmp(a, 'Inlet1D')
+                    d.domainID = ctFunc('inlet_new');
+                elseif strcmp(a, 'Surf1D')
+                    d.domainID = ctFunc('surf_new');
+                elseif strcmp(a, 'Symm1D')
+                    d.domainID = ctFunc('symm_new');
+                elseif strcmp(a, 'Outlet1D')
+                    d.domainID = ctFunc('outlet_new');
+                elseif strcmp(a, 'OutletRes')
+                    d.domainID = ctFunc('outletres_new');
+                else
+                    error('Not enough arguments for that job number');
+                end
```

Comment:
Should be replaced by factory constructors in #1445. The signature is
```c
int domain_new(const char* type, int i, const char* id);
```
where the integer holds a handle to a `Solution`. There are use example in `test/clib/test_ctonedim.cpp`